core/src/test/java/io/methvin/watchservice/DirectoryWatcherTest.java

```diff
@@ -148,8 +148,7 @@ protected void runWatcher(Path directory, WatchService watchService) throws Exce
     List<FileSystemAction> actions = fileSystem.actions();
 
     TestDirectoryChangeListener listener = new TestDirectoryChangeListener(directory, actions);
-    DirectoryWatcher watcher = new DirectoryWatcher(Collections.singletonList(directory), listener, watchService);
-
+    DirectoryWatcher watcher = new DirectoryWatcher(Collections.singletonList(directory), listener, watchService, Boolean.TRUE);
```

Comment:
`true`. Also did you try running the tests with a value of `false` here? It would be interesting to see which tests end up with duplicate events.

Reply:
I think this depends largely on the OS. In case of Win10, it's usually update events that show up multiple times. E.g. in this test, for file one.txt instead of only create and delete, we have: create, update and delete events.